cloudevents/sdk/marshaller.py

```diff
@@ -55,8 +55,18 @@ def FromRequest(self, event: event_base.BaseEvent,
         :return: a CloudEvent
         :rtype: event_base.BaseEvent
         """
+        if not isinstance(data_unmarshaller, typing.Callable):
+            raise exceptions.InvalidDataUnmarshaller()
+
+        content_type = headers.get(
+            "content-type", headers.get("Content-Type"))
+
         for _, cnvrtr in self.__converters.items():
```

Comment:
This is treating `self.__converters` as an unordered list (despite having been passed as an ordered list on line 102. `dict` randomization will mean that binary mode will sometimes be selected for a structured-mode request, which will cause test flakiness. (Discovered this by accident when re-running tests after fixing a lint error.)

Reply:
(I fixed this in https://github.com/cloudevents/sdk-python/pull/9)